pkg/webhooks/jobset_webhook_test.go

```diff
@@ -1097,10 +1120,8 @@ func TestValidateUpdate(t *testing.T) {
 			newObj := tc.js.DeepCopyObject()
 			oldObj := tc.oldJs.DeepCopyObject()
 			_, err = webhook.ValidateUpdate(context.TODO(), oldObj, newObj)
-			if tc.want != nil {
-				assert.True(t, strings.Contains(err.Error(), tc.want.Error()))
-			} else {
-				assert.Nil(t, err)
+			if diff := cmp.Diff(tc.want, err, cmpopts.IgnoreFields(field.Error{}, "BadValue")); diff != "" {
```

Comment:
Why do we need to ignore the BadValue field here

Reply:
We don't need to ignore, tentatively I ignored it because I got a bigger diff for the test case: "replicated_jobs_are_immutable", and it wasn't asserted before:
```diff
            - 		BadValue: string(""),
            + 		BadValue: []v1alpha2.ReplicatedJob{
            + 			{
            + 				Name:     "test-jobset-replicated-job-0",
            + 				Template: v1.JobTemplateSpec{Spec: v1.JobSpec{...}},
            + 				Replicas: 2,
            + 			},
            + 			{
            + 				Name:     "test-jobset-replicated-job-1",
            + 				Template: v1.JobTemplateSpec{Spec: v1.JobSpec{...}},
            + 				Replicas: 1,
            + 			},
            + 		},
```
I'm happy either way, WDYT?

Reply:
The last commit pushed contains validation of the BadValue. PTAL.

Reply:
Yeah I think the bad value can be helpful to see what value was mutated in the immutable construct, so it's easier to find where the mutation is coming from in the code and fix it.